File: nöbetim/src/main/resources/js/services/HospitalService.js
/**
 * @author emirhan yasin cetin
 */
(function () {

    angular.module('nobetim').factory('HospitalService', HospitalService);

    HospitalService.$inject = ['$http', 'urlMapper'];

    function HospitalService($http, urlMapper) {
        return{

            getAllHospitals : function () {
                var httpPromise = $http.get(urlMapper.GET_ALL_HOSPITALS);
                return httpPromise;
            },

            getDepartments : function (hospitalName) {
                var data = {
                    hospitalName: hospitalName
                };

                var httpPromise = $http.get(urlMapper.GET_DEPARTMENTS, data);
                return httpPromise;
            }

        }
    }
}());

File: nöbetim/src/main/java/nobetim/Service/HospitalService.java
package nobetim.Service;

import nobetim.core.Department;
import nobetim.core.Hospital;

import java.util.List;

/**
 * @author emrhan yasin cetin
 */
public interface HospitalService {

    List<Hospital> getHospitals();
    List<String> getHospitalNames();
    void setHospital(Hospital hospital);
    Hospital getHospitalByName(String hospitalName);
    void updateHospital(Hospital hospital);
    List<Department> getDepartments(String hospitalName);
}
